Source/WebCore/platform/graphics/mac/controls/SwitchMacUtilities.mm

```diff
@@ -49,7 +49,22 @@ static IntSize cellSize(NSControlSize controlSize)
     return sizes[controlSize];
 }
 
-static IntOutsets cellOutsets(NSControlSize controlSize)
+static FloatSize logicalCellSize(NSControlSize controlSize, const ControlStyle& style)
+{
+    auto logicalSize = cellSize(controlSize);
+    if (style.states.contains(ControlStyle::State::VerticalWritingMode))
+        logicalSize = logicalSize.transposedSize();
+    logicalSize.scale(style.zoomFactor);
+    return logicalSize;
+}
+
+static IntOutsets cellOutsets(NSControlSize)
+{
+    ASSERT_NOT_IMPLEMENTED_YET();
+    return { };
+}
+
+static IntOutsets logicalCellOutsets(NSControlSize controlSize, bool isVertical)
 {
     static const IntOutsets margins[] =
```

Comment:
Can we keep this in `cellOutsets` and call that method here?

Reply:
Yeah we could, it just wasn't clear what utility that would provide. There's no code calling into cellOutsets from a super class.

Reply:
I'd eventually want to call it to avoid the duplicate cell outset definitions used in the Web process side. Since we already have it implemented, and the code itself is still useful, I would keep it.